File: scripts/seed.py
#!/usr/bin/env python3
"""Seed books into Elise's account via Convex CLI.

Usage:
  python3 scripts/seed.py <userId>
  python3 scripts/seed.py k578mmwdrkg80jzzj0j36n2at57y0h37
"""
import json, subprocess, sys

if len(sys.argv) < 2:
    print("Usage: python3 scripts/seed.py <userId>")
    print("Get userId with: npx convex run users:getProfile")
    sys.exit(1)

user_id = sys.argv[1]

with open("all-books.json", "r") as f:
    books = json.load(f)

payload = json.dumps({"userId": user_id, "books": books})
print(f"📚 Seeding {len(books)} books for user {user_id}...")

result = subprocess.run(
    ["npx", "convex", "run", "books:seedBooks"],
    input=payload,
    capture_output=True,
    text=True,
    cwd=".",
)

print(result.stdout)
if result.returncode != 0:
    print(result.stderr)
    sys.exit(1)

print("✅ Done!")

File: scripts/seed.mjs
#!/usr/bin/env node
// Import books to Elise's account
// Usage: node scripts/seed.mjs <userId>
// Example: node scripts/seed.mjs k578mmwdrkg80jzzj0j36n2at57y0h37

import { readFileSync, writeFileSync } from "fs";
import { execSync } from "child_process";
import { tmpdir } from "os";
import { join } from "path";

const userId = process.argv[2];
if (!userId) {
  console.error("Usage: node scripts/seed.mjs <userId>");
  console.error("Get userId from: npx convex run users:getProfile");
  process.exit(1);
}

const books = JSON.parse(readFileSync("all-books.json", "utf-8"));
console.log(`Seeding ${books.length} books for user ${userId}...`);

const payload = JSON.stringify({ userId, books });

// Write to temp file to avoid shell quoting issues with Unicode
const tmpFile = join(tmpdir(), "convex-seed.json");
writeFileSync(tmpFile, payload, "utf-8");

try {
  execSync(`cat "${tmpFile}" | npx convex run books:seedBooks`, {
    stdio: "inherit",
    cwd: process.cwd(),
  });
  console.log("✅ Done!");
} catch (e) {
  console.error("❌ Failed");
  process.exit(1);
}
